app/assets/javascripts/ui_components/modal.coffee

```diff
@@ -0,0 +1,13 @@
+$ ->
+  _.each $('.ui-components-modal'), (modal) ->
+    $modal = $ modal
+    trigger = $($modal.attr('data-trigger'))
+    return if trigger.length == 0
+    $(trigger).click (e) ->
+      url = $modal.attr('data-url')
+      if url && url.length > 0
+        $modal.find('.modal-body').load url, ->
+          $modal.one('shown.bs.modal', (e) -> $modal.trigger('uic:domchange'))
```

Comment:
I don't get it. What is a "state where you can use it", and how can the markdown-render script not access the data before?